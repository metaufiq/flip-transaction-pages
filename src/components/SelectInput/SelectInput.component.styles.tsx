import { StyleSheet } from "react-native";
import { colors } from "../../configs/themes";

const styles = StyleSheet.create({
  centeredView: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: 'rgba(52, 52, 52, 0.8)',
    marginTop: 22
  },
  modalView: {
    margin: 20,
    backgroundColor: "white",
    borderRadius: 20,
    padding: 35,
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5
  },
  label: {
    color: colors.primary
  },
  optionContainer:{
    padding: 10
  }
})

export default styles;