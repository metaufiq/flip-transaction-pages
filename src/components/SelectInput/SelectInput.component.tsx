import React, { Fragment, useState } from "react";
import { Modal, Pressable, Text, View } from "react-native";

import styles from "./SelectInput.component.styles";
import { Props, SetShowModal } from "./SelectInput.component.types";

const _onShowModal = (setShowModal:SetShowModal)=>()=>{
  setShowModal(true)
}

const _onSelectOption = (props: Props, value:string, setShowModal:SetShowModal)=>()=>{
  props.onChange(value)
  setShowModal(false)
}

const SelectInput = (props: Props)=>{

  const [showModal, setShowModal] = useState(false);
  return(
    <Fragment>
      <Modal
        animationType="slide"
        transparent
        visible={showModal}
      >
        <View style={styles.centeredView}>
          <View style={styles.modalView}>
                {props.options.map(option=>(
                  <Pressable onPress={_onSelectOption(props,option.value, setShowModal)}>
                    <View style={styles.optionContainer}>
                      <Text>{option.label}</Text>
                    </View>
                  </Pressable>
                ))}
          </View>
        </View>
      </Modal>
      <Pressable onPress={_onShowModal(setShowModal)}>
        <Text style={styles.label}>{props.label}</Text>
      </Pressable>
    </Fragment>
  )
}

export default SelectInput;