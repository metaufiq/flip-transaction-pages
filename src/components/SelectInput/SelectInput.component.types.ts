import { SetState, VoidFunction } from "../../../index.types"

export type Option = {
  value: any,
  label: string,
}

export type Props = {
  options: Option[],
  label: string,
  onChange: (value:string)=>void,
  value?: string
}

export type SetShowModal = SetState<boolean>