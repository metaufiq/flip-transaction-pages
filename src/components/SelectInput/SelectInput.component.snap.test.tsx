import { assertSnapshots } from "../../../test";
import SelectInput from ".";
import { Props } from "./SelectInput.component.types";

jest.mock('react', ()=>({
  ...jest.requireActual('react'),
  useState: jest.fn(()=>[false, jest.fn()])
}))

describe('SelectInputSnapshot', () => {

  const mockProps:Props = {
    onChange:()=>{},
    label: 'label',
    options: [{
      label: 'label',
      value: 'value'
    }]
  };

  const configs = [
    {
      props: mockProps,
      desc: 'should render select input'
    }
  ];

  assertSnapshots(SelectInput, configs);
});
