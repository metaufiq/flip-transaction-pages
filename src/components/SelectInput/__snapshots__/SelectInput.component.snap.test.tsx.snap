// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInputSnapshot should render select input 1`] = `
Array [
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  />,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        Object {
          "color": "#fd6542",
        }
      }
    >
      label
    </Text>
  </View>,
]
`;
